### PR TITLE
Bookstore: create and remove books

### PR DESCRIPTION
**Pull request by Ntwali-Josue**
### In this Pull Request, I: 

- Import `createStore`.
- Import `combineReducers`.
- Import `applyMiddleware` - To use some handy Redux middleware, like `redux-logger.`
- Import  `Reducers`, used to write in /src/redux/books directory.
- Write the book's actions and reducer (in /src/redux/books/books.js) using the duck's pattern.
- Implement the add and remove book functionalities.

### CODE REVIEW

**Review by mke2111 — CHANGES_REQUESTED** [review]
### Hey @Ntwali-Josue, 

**This is a great project you have going on, well-done lad**. 

   
**However, there are a few sections to improve before your project is approved.** 🔴 
<hr>

## Required Changes ♻️
_Check the inline comments under the review for sections to improve._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ntwali-Josue** [comment]
Hey @mke2111, thanks for reaching out to review my pull request. However, I am not being able to see the changes that you required, I mean the inline reviews that you are talking about. Can you please look into this and help me access them? thanks.

File: src/components/AddBook.js [hunk 1]
```diff
@@ -1,45 +1,78 @@
...
+    e.preventDefault();
+    if (inputRef.current[0].value && inputRef.current[1].value) {
+      const newBook = {
+        title: inputRef.current[0].value,
+        author: inputRef.current[1].value,
+      };
+      submitBook(newBook);
+      inputRef.current[0].value = '';
+      inputRef.current[1].value = '';
+    }
```
Comment (mke2111) [inline src/components/AddBook.js:29]: - [ ] For each book before submission, it is better to save the book data along with a unique `id` for each book. 
- Generate a unique `id` for each book to the `newBook` object before submitting the new book .

Comment (Ntwali-Josue) [reply src/components/AddBook.js:29]: Hey @mke2111, thanks for mentioning this out. I am of course saving the data with a unique `id` but I do this using the `submitBook` function. You can see that  [here](https://github.com/Ntwali-Josue/bookstore/pull/2/files#diff-38522ffb90ca92d09944ba8afd3cead7813b55c6bf5c06aafb312f4c48de7e54R13-R22) I generate a unique `id` to each added book before storing it, and that's what you are trying to help me do, while I've done it already. Can you please check again using the above link? Thanks.

Comment (mke2111) [reply src/components/AddBook.js:29]: I had seen that you were using the `submitBookToStore` to save a new book with a generated `id`. But you are using the `submitBook`. For this milestone's requirement, you are required to generate the id while submitting the book from the form.

Comment (Ntwali-Josue) [reply src/components/AddBook.js:29]: Oh, okay thanks for letting me know this 👍🏾 . I am going to work on it. 👨🏾‍💻

**Review by mke2111 — APPROVED** [review]
### Hi @Ntwali-Josue, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
**Congratulations! 🎉** 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Ntwali-Josue** [comment]
Thanks @mke2111 for approving my project 🕺🏾

**Pull request merged**